internal/backuprestore/backup.go

```diff
@@ -310,29 +295,37 @@ func (h *BRHandler) ExportOadpConfigurationToDir(ctx context.Context, toDir, oad
 			}
 
 			secretName := creds["name"].(string)
-			if secretName == defaultStorageSecret {
-				// default storage secret has been exported already, continue
-				continue
-			}
+			secrets = append(secrets, secretName)
+		}
+	}
 
-			storageSecret := &corev1.Secret{}
-			if err := h.Get(ctx, types.NamespacedName{
-				Name:      secretName,
-				Namespace: oadpNamespace,
-			}, storageSecret); err != nil {
-				return err
-			}
+	if len(secrets) == 0 {
+		return nil
+	}
+	// Create the directory for secrets
+	if err := os.MkdirAll(filepath.Join(toDir, oadpSecretPath), 0o700); err != nil {
+		return err
+	}
 
-			storageSecret.SetUID("")
-			storageSecret.SetResourceVersion("")
+	// Write secrets
+	for _, secretName := range secrets {
+		storageSecret := &corev1.Secret{}
+		if err := h.Get(ctx, types.NamespacedName{
+			Name:      secretName,
+			Namespace: oadpNamespace,
+		}, storageSecret); err != nil {
+			return err
+		}
 
-			filePath := filepath.Join(toDir, oadpSecretPath, secretName+".yaml")
-			if err := writeSecretToFile(storageSecret, filePath); err != nil {
-				return err
-			}
+		storageSecret.SetUID("")
+		storageSecret.SetResourceVersion("")
+
+		filePath := filepath.Join(toDir, oadpSecretPath, secretName+".yaml")
+		if err := writeSecretToFile(storageSecret, filePath); err != nil {
```

Comment:
There's a MarshalToFile function in utils (https://github.com/openshift-kni/lifecycle-agent/blob/main/utils/utils.go#L27) that I think can replace writeSecretToFile. We should probably change the mod to 0o600, though.
This can be a separate PR